org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/XMLTextDocumentService.java

```diff
@@ -215,6 +216,11 @@ private XMLFormattingOptions getFormattingSettings(String uri) {
 	@Override
 	public CompletableFuture<List<Either<SymbolInformation, DocumentSymbol>>> documentSymbol(
 			DocumentSymbolParams params) {
+		
+		if(!sharedSettings.symbolSettings.isEnabled()) {
+			return CompletableFuture.completedFuture(null);

```

Comment:
CompletableFuture.completedFuture(Collections.emptyList());

else, you might break some clients